homeassistant/components/mobile_app/webhook.py

```diff
@@ -295,10 +298,11 @@ async def webhook_render_template(hass, config_entry, data):
     resp = {}
     for key, item in data.items():
         try:
-            tpl = item[ATTR_TEMPLATE]
-            attach(hass, tpl)
+            tpl = template.Template(item[ATTR_TEMPLATE])
+            tpl.ensure_valid()
+            tpl.hass = hass
             resp[key] = tpl.async_render(item.get(ATTR_TEMPLATE_VARIABLES))
-        except TemplateError as ex:
+        except Exception as ex:  # pylint: disable=broad-except
```

Comment:
@bdraco this is a little weird I think. We need to catch all exceptions because template `{{ now() | random }}` raises a TypeError that crashes the websocket connection (500 error) and also here had to be caught with a generic error. Should we catch all exceptions that come out of render and just wrap it in TemplateError ?

Reply:
Wrapping `.render` in `template.py` seems like it will also fix some other bugs where templates stop updating because the exception isn't trapped.

I'm doing testing with it now.

Reply:
It also looks like `{{ now() | rando }}` is untrapped.

Reply:
I've found a few things.  I'll open a new PR to fix them, when its done trapping `except TemplateError as ex` will do what was originally expected here.

Reply:
Fixed in https://github.com/home-assistant/core/pull/41719